worker/src/main/java/gov/cms/ab2d/worker/processor/PatientClaimsProcessorImpl.java

```diff
@@ -47,19 +65,85 @@
      */
     @Trace(metricName = "EOBRequest", dispatcher = true)
     @Async("patientProcessorThreadPool")
-    public Future<EobSearchResult> process(PatientClaimsRequest request) {
+    public Future<ProgressTrackerUpdate> process(PatientClaimsRequest request) {
+        ProgressTrackerUpdate update = new ProgressTrackerUpdate();
         final Token token = request.getToken();
         token.link();
-
+        List<CoverageSummary> patients = request.getCoverageSummary();
+        FhirVersion fhirVersion = request.getVersion();
         try {
-            List<IBaseResource> eobs = getEobBundleResources(request);
-            EobSearchResult result = new EobSearchResult(request.getJob(), request.getContractNum(), eobs);
-            return new AsyncResult<>(result);
+            File file = null;
+            String anyErrors = null;
+            try (BeneficiaryStream stream = new BeneficiaryStream(request.getJob(), request.getEfsMount(), DATA,
+                    this.streamingDir, this.finishedDir, this.bufferSize)) {
+                file = stream.getFile();
+                logManager.log(new FileEvent(request.getOrganization(), request.getJob(), stream.getFile(), FileEvent.FileStatus.OPEN));
+                for (CoverageSummary patient : patients) {
+                    List<IBaseResource> eobs = getEobBundleResources(request, patient, update);
+                    anyErrors = writeOutResource(fhirVersion, update, eobs, stream);
+                    update.incPatientProcessCount();
+                }
+            } finally {
+                logManager.log(new FileEvent(request.getOrganization(), request.getJob(), file, FileEvent.FileStatus.CLOSE));
+            }
+            if (anyErrors != null && anyErrors.length() > 0) {
+                File errorFile = null;
+                try (BeneficiaryStream stream = new BeneficiaryStream(request.getJob(), request.getEfsMount(), ERROR,
+                        this.streamingDir, this.finishedDir, this.bufferSize)) {
+                    errorFile = stream.getFile();
+                    logManager.log(new FileEvent(request.getOrganization(), request.getJob(), stream.getFile(), FileEvent.FileStatus.OPEN));
+                    stream.write(anyErrors);
+                } catch (IOException e) {
+                    log.error("Cannot log error to error file");
+                } finally {
+                    logManager.log(new FileEvent(request.getOrganization(), request.getJob(), errorFile, FileEvent.FileStatus.CLOSE));
+                }
+            }
```

Comment:
Not found of so much nesting.  I suggest (not require) a refactor to method.

Less nesting improves clarity and allows one more flexibility in exit paths within a method.

Reply:
I'm not sure nested is the exact issue, maybe just too long a method.

Reply:
There is a try nested within a try.